### PR TITLE
Add support for MachineConfig and add Type Definitions

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Success! https://github.com/NullVoxPopuli/emberclear/pull/803

image

it figured out the types of the guard args! (Context)

### CODE REVIEW

File: addon/usables/use-machine.js [hunk 1]
```diff
@@ -21,6 +22,7 @@ export class InterpreterService {
 
   setup() {
     this.service = interpret(this.machine, {
+      devTools: DEBUG,
```
Comment (LevelbossMike) [inline addon/usables/use-machine.js:25]: I'm not aware how `DEBUG` works. Is this true in development and test? Or do users need to set this manually via an env-variable?

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import {
+  StateMachine, State,
+  MachineConfig,
+  Interpreter, EventType, EventObject,
+  InterpreterOptions
+} from "xstate";
+
+// NOTE: there is a lot of repetition here :(
+// Repetition will be less when ember-statecharts _is_ TS?
```
Comment (LevelbossMike) [inline index.d.ts:9]: :+1: I plan to move this over to TS asap

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  send: Send<Context, Schema, Events>;
+  withConfig: (...args: WithConfigParams<Context, Schema, Events>) => InterpretedUsable<Context, Schema, Events>;
+  withContext: (...args: WithContextParams<Context, Schema, Events>) => InterpretedUsable<Context, Schema, Events>;
+}
+
+type UsableStatechart<Context, Schema, Events extends EventObject> =
+  | MachineConfig<Context, Schema, Events>
+  | StateMachine<Context, Schema, Events>
+
+export function useMachine<Context, Schema, Events extends EventObject>(
```
Comment (LevelbossMike) [inline index.d.ts:26]: Nice 👍

File: addon/usables/use-machine.js (hunk 1 above)
Comment (NullVoxPopuli) [reply addon/usables/use-machine.js:25]: It's true during dev and test, and false in production.

glimmer/env isn't even a real package, I don't think but something configured by babel during build

File: addon/usables/use-machine.js [hunk 4]
```diff
@@ -21,6 +23,8 @@ export class InterpreterService {
...
+      ...this.interpreterOptions,
```
Comment (NullVoxPopuli) [inline addon/usables/use-machine.js:27]: if desired, someone could make devTools permanently on, or off (or pass other options)

**Pull request merged**